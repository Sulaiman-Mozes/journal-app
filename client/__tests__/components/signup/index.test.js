import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Signup from '../../../components/signup';

describe('Provider and Signup', () => {
  const props = {
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    fields: { email: '', password: '', username: '' },
  };
  const wrapper = shallow(<Signup {...props} />);

  it('renders <Provider/> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
