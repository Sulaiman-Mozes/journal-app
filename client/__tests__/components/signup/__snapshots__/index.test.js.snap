// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider and Signup renders <Provider/> correctly 1`] = `
<div
  className="padding-top form-container padding-bottom"
>
  <form
    className=" padding-top text-center border border-light p-5"
    onSubmit={[MockFunction]}
  >
    <p
      className="h4 mb-4"
    >
      Sign up
    </p>
    <input
      className="form-control mb-4"
      id="defaultRegisterFormUsername"
      name="username"
      onChange={[MockFunction]}
      placeholder="Username"
      required={true}
      type="text"
      value=""
    />
    <input
      className="form-control mb-4"
      id="defaultRegisterFormEmail"
      name="email"
      onChange={[MockFunction]}
      placeholder="E-mail"
      required={true}
      type="email"
      value=""
    />
    <input
      aria-describedby="defaultRegisterFormPasswordHelpBlock"
      className="form-control"
      id="defaultRegisterFormPassword"
      name="password"
      onChange={[MockFunction]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <small
      className="form-text text-muted mb-4"
      id="defaultRegisterFormPasswordHelpBlock"
    >
      At least 8 characters and 1 digit
    </small>
    <button
      className="btn btn-info my-4 btn-block"
      type="submit"
    >
      Sign up
    </button>
    <hr />
    <p>
      By clicking
      <em>
         Sign up 
      </em>
      you agree to our
      <NavLink
        target="_blank"
        to="/"
      >
         terms of service
      </NavLink>
    </p>
  </form>
</div>
`;
